### PR TITLE
MOTOR-708: Update Changelog for Motor 2.4

### PR DESCRIPTION
**Pull request by guanlinzhou**

### CODE REVIEW

File: doc/changelog.rst [hunk 1]
```diff
@@ -6,6 +6,15 @@ Changelog
 Motor 2.4
 ---------
 
+Motor 2.4 adds support for client-side field-level encryption
```
Comment (prashantmital) [inline doc/changelog.rst:9]: We should link to the newly added CSFLE examples here like we do in this changeling entry for PyMongo: https://pymongo.readthedocs.io/en/stable/changelog.html#changes-in-version-3-10-0

**Review by prashantmital — CHANGES_REQUESTED** [review]
We also need to mention that this release adds support for Python 3.9 

Additionally, can you update the https://motor.readthedocs.io/en/stable/requirements.html#compatibility-matrix to include the 2.4 version in both tables.

**Comment by guanlinzhou** [comment]
> We also need to mention that this release adds support for Python 3.9
> 
> Additionally, can you update the https://motor.readthedocs.io/en/stable/requirements.html#compatibility-matrix to include the 2.4 version in both tables.

I think I did mention Python 3.9 support in the changelog already

**Comment by ShaneHarvey** [comment]
Feel free to add your name to `doc/contributors.rst` too!

File: doc/changelog.rst [hunk 2]
```diff
@@ -13,6 +13,7 @@ New Features:
 
 - Added the :class:`motor.motor_asyncio.AsyncIOMotorClientEncryption` class,
 with the same interface as the corresponding PyMongo class.
+See :doc:`examples/encryption` for examples.
```
Comment (prashantmital) [inline doc/changelog.rst:16]: Does this render correctly? I think you need to manually add space before continuing the bullet on the next line. See the other multi-line changelog entries for an example. You could also try to build the docs yourself and see if that works.

**Review by prashantmital — CHANGES_REQUESTED** [review]

File: doc/changelog.rst (hunk 2 above)
Comment (guanlinzhou) [reply doc/changelog.rst]: ah you're right, fixed!

**Review by prashantmital — APPROVED** [review]

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**